client/lib/i18n-utils/switch-locale.js

```diff
@@ -65,8 +65,22 @@ export function getLanguageFileUrl( localeSlug, fileType = 'json', languageRevis
 	return typeof revision === 'number' ? fileUrl + `?v=${ revision }` : fileUrl;
 }
 
-function setLocaleInDOM( localeSlug, isRTL ) {
-	document.documentElement.lang = localeSlug;
+function getHtmlLangAttribute() {
+	// translation of this string contains the desired HTML attribute value
+	const slug = i18n.translate( 'html_lang_attribute' );
+
+	// Hasn't been translated? Likely the default `en` locale. Return the default i18n slug.
+	if ( slug === 'html_lang_attribute' ) {
+		return i18n.getLocaleSlug();
```

Comment:
Could we rather use the `localeSlug` here? We still have some languages that have it translated to `html_lang_attribute` despite being not English.

Reply:
Yes, I can change that! Do you have an example of such a language that doesn't translate `html_lang_attribute`? It will help me test the patch.

Reply:
`ps` is an example language: http://widgets.wp.com/languages/calypso/ps-v1.1.json

Reply:
@akirk after looking at this more closely, I believe `i18n.getLocaleSlug()` is already correct. It returns the `localeSlug` of the locale that's just been set inside `i18n-calypso` a few moments before.

It doesn't return the default language, i.e., `en`. Maybe it's the comment that is confusing and should be updated?

Reply:
Yes, let's remove "default", I guess because of it I misread this as a `getDefaultLocaleSlug()`. Sorry for the false alarm!

Reply:
Comment updated and clarified in https://github.com/Automattic/wp-calypso/pull/38196/commits/7dcde4a4ede267c93778c05791077044956326b0

Ready for final review.